app/workers/charts/epoch_statistic.rb

```diff
@@ -7,9 +7,26 @@ def perform
       latest_epoch_number = ::EpochStatistic.order(epoch_number: :desc).pick(:epoch_number) || 0
       target_epoch_number = latest_epoch_number + 1
       block_in_target_epoch = Block.where(epoch: target_epoch_number).first
-      if block_in_target_epoch.present? && Block.where(epoch: target_epoch_number).count == block_in_target_epoch.length
-        Charts::EpochStatisticGenerator.new(target_epoch_number).call
-      end
+
+      return unless block_in_target_epoch
+
+      blocks_count = Block.where(epoch: target_epoch_number).count
+      return unless blocks_count == block_in_target_epoch.length
+
+      epoch_statistic = ::EpochStatistic.find_or_create_by(epoch_number: target_epoch_number)
```

Comment:
I found these code and next `fill_epoch_statistic `method code  were same.So maybe we can replace  `Charts::EpochStatisticGenerator` 's code with your code.